.pre-commit-config.yaml

```diff
@@ -42,17 +42,28 @@ repos:
     stages: [manual]
 
 - repo: https://github.com/pre-commit/mirrors-mypy
-  rev: v0.812
+  rev: v0.902
   hooks:
   - id: mypy
     exclude: ^(bin/|cibuildwheel/resources/).*py$
-    additional_dependencies: [packaging]
-    args: ["--python-version=3.6", "--ignore-missing-imports", "--scripts-are-modules"]
+    args: ["--python-version=3.6", "--scripts-are-modules"]
+    additional_dependencies:
+      - packaging
+      - types-jinja2
+      - types-certifi
+      - types-toml
   - id: mypy
     name: mypy 3.7+ on bin/
     files: ^bin/.*py$
-    additional_dependencies: [packaging]
-    args: ["--python-version=3.7", "--ignore-missing-imports", "--scripts-are-modules"]
+    args: ["--python-version=3.7", "--scripts-are-modules"]
+    additional_dependencies:
+      - packaging
+      - rich
+      - types-jinja2
+      - types-pyyaml
+      - types-click
+      - types-requests
+      - types-toml
```

Comment:
Should we add these to the `dev` requirements in setup.py too?

Reply:
Sure, we can. I'd normally see mypy as something you really should use either nox or pre-commit for, not something you'd set up in your development environment, but it shouldn't hurt. There's also @asottile's `types-all`.

I'm moving these to a "mypy" extra that gets included in dev.

Reply:
Brilliant, thanks. That's helpful to me, I do sometimes run `mypy .` at a shell, because pre-commit is more interested in the staged files than the working copy.

Reply:
```console
$ pre-commit run -a
```

This will run on all files and will ignore the staging area.

Reply:
```console
$ pre-commit run -a mypy
```

If you only want the mypy check. :)

Reply:
ah yeah that's pretty handy!